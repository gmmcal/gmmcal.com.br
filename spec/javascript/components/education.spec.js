import { mount } from '@vue/test-utils'
import VueMarkdown from '@/components/markdown'
import Education from '@/components/education'
import create from '../settings/utils'

describe('Education', () => {
  let props, wrapper

  beforeEach(() => {
    props = {
      propsData: {
        education: create('education')
      }
    }

    wrapper = mount(Education, props)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Content', () => {
    it('has an end date', () => {
      const el = wrapper.find('.year')
      expect(el.text()).toBe('Jan / 2016')
    })

    it('has an institution', () => {
      const el = wrapper.find('h4')
      expect(el.text()).toBe('Some awesome university')
    })

    it('has a course', () => {
      const el = wrapper.find('h5')
      expect(el.text()).toBe('Computer science')
    })

    it('has a description', () => {
      const el = wrapper.find('.description')
      expect(el.text()).toBe('Lorem ipsum nulus dolor')
    })

    describe('without end date', () => {
      beforeEach(() => {
        props = {
          propsData: {
            education: create('education', { end_date: null })
          }
        }

        wrapper = mount(Education, props)
      })

      it('has Currently text', () => {
        const el = wrapper.find('.year')
        expect(el.text()).toBe('Currently')
      })
    })
  })

  describe('Child components', () => {
    it('includes VueMarkdown', () => {
      const child_component = wrapper.find(VueMarkdown)
      expect(child_component.isVueInstance()).toBeTruthy()
    })
  })
})
