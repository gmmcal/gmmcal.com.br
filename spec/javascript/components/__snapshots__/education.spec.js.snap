// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education renders correctly 1`] = `
<div class="row education-details">
  <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
    <div class="year">Jan / 2016</div>
  </div>
  <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
    <div class="arrowpart"></div>
    <div class="ex-con">
      <h4>Some awesome university</h4>
      <h5>Computer science</h5>
      <div class="description">
        <p>Lorem ipsum nulus dolor</p>
      </div>
    </div>
  </div>
</div>
`;
